### DISCUSSION TITLE
EXORing Two 2-char Hex Values  M5Stack UIFlow  or MicroPython

### DISCUSSION THREAD

**Discussion by ashimotok0** [micropython/micropython#10467] [General]
I need to EXOR Two 2-char Hex Values using  M5Stack UIFlow or MicroPython. Any ideas please?

**Comment by 2dof** [comment]
solutions on stackoverflow for python also valid for micropython:  
https://stackoverflow.com/questions/11119632/bitwise-xor-of-hexadecimal-numbers

example:
>>> print(hex(0x12ef ^ 0xabcd))

for mote info about operators:
https://note.nkmk.me/en/python-bit-operation/

**Comment by peterhinch** [comment]
If the result is to be a 'bytes' instance there are various ways. Here is one approach:
```python
>>> a = b'\x01\x02'
>>> b = b'\x03\x10'
>>> int.to_bytes(int.from_bytes(a, 'little') ^ int.from_bytes(b, 'little'), 2, 'little')
b'\x02\x12'
```
Or (may be more efficient for large objects):
```python
>>> a = b'\x01\x02'
>>> b = b'\x03\x10'
>>> bytes(map(lambda x, y: x^y, a, b))
b'\x02\x12'
>>> 
```
